src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -769,6 +770,13 @@ object AppDefinition extends GeneralPurposeCombinators {
     }
   }
 
+  private val complyWithMesosHealthChecks: Validator[AppDefinition] =
+    conditional[AppDefinition](appDef =>
+      (appDef.healthChecks.count(_.isInstanceOf[MesosHealthCheck]) -
+        appDef.healthChecks.count(_.isInstanceOf[MesosCommandHealthCheck])) > 0) {
+      featureEnabled(Features.MESOS_HEALTHCHECKS)
```

Comment:
Does the libmesos version has anything to do with communication. I can use libmesos from previous version and my messages will be correctly transported. Libmesos has no correlation with real Mesos version. Instead of this we should query Mesos master to get it's version.